src/crimson/osd/ops_executer.cc

```diff
@@ -27,6 +27,8 @@ namespace {
 
 namespace crimson::osd {
 
+thread_local OSDOpParams* OpsExecuter::osd_op_params = (OSDOpParams*) new char[sizeof(OSDOpParams)];
```

Comment:
Why switch it to thread_local?

Reply:
I thought, since osd_op_params is a static member of OpsExecuter and there may be multiple instances of OpsExecuter accessed by different threads, there could be a chance that osd_op_params are accessed by different threads simultaneously, which would be a problem. Am I right?

Reply:
Or, it shouldn't be a static member?

Reply:
I wouldn't have expected it to be static either -- perhaps I misread something.  Was there a reason you wanted a single instance/thread?  Note that you need to consider the case where multiple reads are in progress concurrently (not in parallel, simply that while one is unscheduled pending a read, another might be running).

Reply:
i concur with @athanatos . to my understanding, op_params is a per transaction information, there is no need to stick it to a *thread*. what's the consideration you allocate in a TLS or a static variable?

Reply:
At first, I thought, since osd_op_params is only used in submit_changes(), maybe we can avoid allocating memory for osd_op_params each time an OpsExecuter is instantiated and reuse the same piece of memory for osd_op_params. But, now, I have doubts, too. Am I considering this right?

Reply:
i think you are trading readability for premature optimization . what's the size of of a `op_params`, and what's the overhead of allocating it on **stack**. also, what's the point of passing by rvalue reference here? can we just pass a `const OSDOpParams&`?

Reply:
I got your point, will change it:-)